proplet/config.go

```diff
@@ -0,0 +1,77 @@
+package proplet
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"os"
+)
+
+// Config holds configuration for the MQTT client.
+type Config struct {
+	BrokerURL     string `json:"broker_url"`
+	Password      string `json:"password"`
+	PropletID     string `json:"proplet_id"`
+	ChannelID     string `json:"channel_id"`
+	RegistryURL   string `json:"registry_url"`
+	RegistryToken string `json:"registry_token"`
+}
+
+// LoadConfig loads and validates the configuration from a JSON file.
```

Comment:
Shouldn't this function be self-explanatory. Only add comments if code is not self explanatory or there is some hidden or complex logic